### PR TITLE
CI: cache dependencies

### PR DESCRIPTION
**Pull request by 678098**

### CODE REVIEW

**Comment by 678098** [comment]
@pniedzielski this is a POC PR, and I believe, we can refine it more. Not sure if the current solution is elegant. Also it increases CI difficulty

**Comment by pniedzielski** [comment]
> @pniedzielski this is a POC PR, and I believe, we can refine it more. Not sure if the current solution is elegant. Also it increases CI difficulty

Right, I see it only deals with normal Ubuntu builds and not Prometheus or Darwin.  Ubuntu is the most important anyway, and slower than Darwin.  I'll take a deeper look tomorrow but I think more elegant will mean more extensive refactoring.  It makes me want to look at what other big projects do for CI with GH Actions.

File: docker/build_deps.sh [hunk 1]
```diff
@@ -37,17 +45,21 @@ configure() {
...
-    bbs_build build -j8
+    if [ "$install_only" = false ]; then
+        bbs_build configure
+        bbs_build build -j8
+    fi
     bbs_build --install=/opt/bb --prefix=/ install
     popd
 }
 
 build_ntf() {
```
Comment (678098) [inline docker/build_deps.sh:56]: This `build_ntf` function causes ntf to rebuild from scratch every time this script is called, no matter if incremental build object exist. This rebuild takes 5 minutes.

`build_bde`, on the other hand, doesn't recompile everything.

If we can get ntf not to rebuild everything, we can even omit this `install_only` flag.

Comment (pniedzielski) [reply docker/build_deps.sh:56]: I'm looking into this now, I will add some comments with what I find.

Comment (pniedzielski) [reply docker/build_deps.sh:56]: By default, ntf's `configure` script cleans build artifacts before it does anything.  When it's passed the `--keep` flag it refrains from doing this.

**Review by pniedzielski — APPROVED** [review]
@678098 I removed the `--install-only` flag in 7fc253a.  Coupled with the `--keep` flag to NTF, it avoids rebuilding BDE and NTF if they are in the cache, without major modifications to `build-deps.sh`.

You should also review it and make sure it was in line with what you suggested, then squash-merge if you'd like.

**Pull request merged**